### PR TITLE
play: fix play service

### PR DESCRIPTION
**Pull request by oliver-sanders**
Fix an issue with a default value in the schema which was causing issues in the UI, may come back to this later, for now easier to remove it (note the end result is the same either way around).

Also fix the response to allow the UI to detect command failure.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Tests (needs some kind of full-system test, don't have time to create a new framework pre-beta)
- [x] No change log entry required (why? un-released)
- [x] No documentation update required.
- [x ] No dependency changes.

### CODE REVIEW

File: cylc/uiserver/schema.py [hunk 1]
```diff
@@ -150,9 +150,7 @@ class Arguments:
                 this cycle point.
             ''')
         )
-        mode = RunMode(
-            default_value=RunMode.Live.value  # type: ignore
-        )
+        mode = RunMode()
```
Comment (oliver-sanders) [inline cylc/uiserver/schema.py:153]: This default value was causing client-server errors, not sure why, removing for the beta.

File: cylc/uiserver/resolvers.py [hunk 2]
```diff
@@ -108,22 +124,13 @@ async def play(workflows, args, workflows_mgr):
...
         except Exception as exc:
             # oh noes, something went wrong, send back confirmation
-            return response.append(
-                [
-                    {
-                        'error': {
-                            'message': str(exc)
-                        }
-                    }
-                ]
```
Comment (oliver-sanders) [inline cylc/uiserver/resolvers.py:118]: Graphene does this templating for us.

File: mypy.ini [hunk 3]
```diff
@@ -3,6 +3,6 @@ python_version = 3.7
 ignore_missing_imports = True
 files = cylc/uiserver
 # don't run mypy on these files directly
-exclude = cylc/uiserver/(websockets/tornado.py|config_defaults.py)
+exclude = cylc/uiserver/(websockets/tornado.py|config_defaults.py|tests/*)
```
Comment (oliver-sanders) [inline mypy.ini:6]: MyPy really doesn't like the compiled protobuf file imported in the tests. Inline ignore comments didn't seem to be enough to suppress the error. Tried copying the cylc-flow mypy config but that didn't fix it so just ignoring the tests for now, can re-instate later.

Comment (kinow) [reply mypy.ini:6]: Did you add the `# type: ignore` in the ... wrong line? I noticed from @MetRonnie that he had to add the comment in the initial line when you had a multi-line import.

If I remove `test/*` from the exclusion in your branch, the command fails. Then this diff fixes it.

```diff
diff --git a/cylc/uiserver/tests/conftest.py b/cylc/uiserver/tests/conftest.py
index 167ba6c..a8a838e 100644
--- a/cylc/uiserver/tests/conftest.py
+++ b/cylc/uiserver/tests/conftest.py
@@ -19,7 +19,7 @@ import inspect
 
 import pytest
 import zmq
-from cylc.flow.data_messages_pb2 import (
+from cylc.flow.data_messages_pb2 import (  # type: ignore
     PbEntireWorkflow, PbWorkflow, PbFamilyProxy)
 from cylc.flow.network import ZMQSocketBase
 
diff --git a/mypy.ini b/mypy.ini
index 20eca2f..b0b4d93 100644
--- a/mypy.ini
+++ b/mypy.ini
@@ -3,6 +3,6 @@ python_version = 3.7
 ignore_missing_imports = True
 files = cylc/uiserver
 # don't run mypy on these files directly
-exclude = cylc/uiserver/(websockets/tornado.py|config_defaults.py|tests/*)
+exclude = cylc/uiserver/(websockets/tornado.py|config_defaults.py)
 # don't run mypy on those files when imported from other modules
 follow_imports = silent

```

**Review by kinow — APPROVED** [review]
Looks good. Just one comment about `mypy`

File: mypy.ini (hunk 3 above)
Comment (MetRonnie) [reply mypy.ini:6]: Yes, it only worked for me when I used `# type: ignore` on the first line of the multiline import

Comment (oliver-sanders) [reply mypy.ini:6]: Tried `type: ignore` but there were problems besides that, managed to get them fixed here https://github.com/cylc/cylc-uiserver/pull/188/commits/0029244360b68328861211bb8ff24ab4a5ab5f8f

**Pull request merged**